UnitySDK/Assets/ML-Agents/Scripts/ICommunicator.cs

```diff
@@ -138,10 +137,15 @@ public interface ICommunicator
         Dictionary<Agent, AgentAction> GetActions(string key);
 
         /// <summary>
-        /// Registers a side channel to the communicator. The side channel will exchange 
+        /// Registers a side channel to the communicator. The side channel will exchange
         /// messages with its Python equivalent.
         /// </summary>
         /// <param name="sideChannel"> The side channel to be registered.</param>
         void RegisterSideChannel(SideChannel sideChannel);
+
+        /// <summary>
+        /// Shut down the communicator and clean up anything that it owns.
+        /// </summary>
+        void Dispose();
```

Comment:
Better name? We already had RpcCommunicator.Dispose.

Reply:
Or just make ICommunicator inherit IDisposable ?

Reply:
Done. Also made Academy IDisposable, and renamed Academy.OnDestroy to Dispose()